version: '3.8'

services:
  db:
    image: postgres:15-alpine
    container_name: splint_postgres_db
    restart: always
    env_file:
      - .env
    environment:
      - POSTGRES_USER=${POSTGRES_USER}
      - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
      - POSTGRES_DB=${POSTGRES_DB}
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - "5433:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${POSTGRES_USER} -d ${POSTGRES_DB}"]
      interval: 5s
      timeout: 5s
      retries: 5

  auth-service:
    build:
      context: ./splint-auth-service
      dockerfile: Dockerfile
    container_name: splint_auth_service
    ports:
      - "8000:8000"
    volumes:
      - ./cert.pem:/app/cert.pem
      - ./key.pem:/app/key.pem
    env_file:
      - .env

  splint-app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: splint_main_app
    ports:
      - "5000:5000"
    volumes:
      - ./server:/app/server
      - ./cert.pem:/app/cert.pem
      - ./key.pem:/app/key.pem
    env_file:
      - .env
    depends_on:
      db:
        condition: service_healthy
      auth-service:
        condition: service_started

volumes:
  postgres_data:
